### PR TITLE
Ports/psoc6 machine pin tests

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
The tests pass as expected for machine pin module. @boramonideep : Soft reset did not help for resetting. I have captured the logs and will share. Is that expected at the moment?

### CODE REVIEW

**Review by boramonideep — CHANGES_REQUESTED** [review]
The `ports/psoc6/mpy_define_mtb_vars` is a static build process variable. Its not to be committed.

**Review by IFX-Anusha — APPROVED** [review]

**Comment by jaenrig-ifx** [comment]
So I just added the complete folder to the tests, instead of just the pin.py. 
Later we should extend to all the relevant tests, but for now...

File: tests/psoc6/pin.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from machine import Pin
+
+# Tests for Output pin
+# Instantiate output pin for led operation
+# Constr
+p1 = Pin("P13_7")
+p1.init(Pin.OUT)
+print(p1)  # Pin:111 or P13_7, Mode=OUT, Pull=None, Value=1
+
+p1.value(0)  # Led glows
```
Comment (jaenrig-ifx) [inline tests/psoc6/pin.py:10]: I don´t want to complicate much more the setup for now, but we could connect 2 pins together and actually check the input against an output automatically. LED would require visual inspection.
That will require a HIL hardware setup documented somewhere.

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
Nice 👍 Only remove the mpy_define_mtb_vars

**Review by jaenrig-ifx — APPROVED** [review]

**Comment by jaenrig-ifx** [comment]
For some reason, the tests on target are failing on the HIL for the pull request. But it runs locally, and it also run with in the push event. So it works.
I have reset the board, but still persists. 
I will have a look it is keeps reproducing.

**Comment by NikhitaR-IFX** [comment]
> The `ports/psoc6/mpy_define_mtb_vars` is a static build process variable. Its not to be committed.

Removed now.

File: tests/psoc6/pin.py (hunk 1 above)
Comment (NikhitaR-IFX) [reply tests/psoc6/pin.py:12]: Yeah i agree.

File: tests/psoc6/pin.py.exp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Pin constructor invoked
+Pin class object instantiated
+init helper function called
+Direction: 1, Drive:6, Value:1
+machine pin print
+Pin:111 or P13_7, Mode=OUT, Pull=None, Value=1
+machine pin call
+machine pin call
+None
+None
```
Comment (jaenrig-ifx) [inline tests/psoc6/pin.py.exp:10]: All this traces are available by default? 
I would consider some of them only relevant for a debugging mode.

Comment (boramonideep) [reply tests/psoc6/pin.py.exp:10]: The first two lines should only be available in debug mode. The switch to enable debug mode is active in the `ports/psoc6` branch and also this branch, which should not be the case. See here: https://github.com/jaenrig-ifx/micropython/blob/e76692e849b54fa5c08cdeddadc7b1641c6b03c9/ports/psoc6/mpconfigport.h#L134

The last two lines (None's) are a return of MPY functions which is defined in their guidelines so those should normally be there irrespective of debug mode active or not.

We should disable the switch `#define MICROPY_LOGGER_DEBUG                    (0)` and have the exp files updated for the tests.

Comment (NikhitaR-IFX) [reply tests/psoc6/pin.py.exp:3]: Yes I agree. I was testing out what happens in debug mode and did not revert. My bad. Done now! Thank you for highlighting.

Comment (jaenrig-ifx) [reply tests/psoc6/pin.py.exp:3]: Understood.
I mean, for test, if you see convenient to have the test in debug mode, we can keep it like that... But by the productive version it should be disabled. 
If that is the case we could pass the flag during compilation instead and use ifndef guards.

Comment (jaenrig-ifx) [reply tests/psoc6/pin.py.exp:3]: mmm yes, but we would have 2 .exp files. That would require to modify the run-tests.py utility.
The only problem with the logger mode enable, is that that is not really the actual program, as all the debugger traces have an impact on performance: memory, timing, code size... 
So, maybe we are adding more complexity than the value we get by running the tests in debug mode?

**Review by bargfred — APPROVED** [review]
Logger debug mode is for debugging, not for testing. If logger debug mode is enclosed in other "#ifdef" clauses (with MPY config parameters), this can become quiet complex, ie many test cases.
But we should have a test case to test the logger itself - but with the final logger. We may want to use the MPY logger in the long run, it was just too tricky to get it up and running, I think. Monideep ? 
I have made progress with the configuration, so we may be able to fix it soon.

File: tests/psoc6/pin.py.exp (hunk 2 above)
Comment (boramonideep) [reply tests/psoc6/pin.py.exp:3]: I agree with @jaenrig-ifx. We should not touch the `run_tests.py` anymore.

**Comment by boramonideep** [comment]
> 

Yes @bargfred. The plan is to use the MPY logger infrastructure for debug logging. Since it is used throughout the MPY modules, we can enable it once most of the vital ones are implemented so that they do not throw errors.

**Review by boramonideep — APPROVED** [review]
Thank you @NikhitaR-IFX. Everything looks great.

**Pull request merged**